dockernet/ts-tests/test/utils.ts

```diff
@@ -0,0 +1,210 @@
+ */
+export const ibcDenom = (
+  paths: {
+    incomingPortId: string;
+    incomingChannelId: string;
+  }[],
+  coinMinimalDenom: string,
+): string => {
+  const prefixes: string[] = [];
+  for (const path of paths) {
+    prefixes.push(`${path.incomingPortId}/${path.incomingChannelId}`);
+  }
+
+  const prefix = prefixes.join("/");
+  const denom = `${prefix}/${coinMinimalDenom}`;
+
+  return "ibc/" + toHex(sha256(toUtf8(denom))).toUpperCase();
+};
+
+/**
+ * Convert a secp256k1 compressed public key to an address
+ *
+ * @param {Uint8Array} pubkey The account's pubkey, should be 33 bytes (compressed secp256k1)
+ * @param {String} [prefix="stride"] The address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function pubkeyToAddress(
+  pubkey: Uint8Array,
+  prefix: string = "stride",
+): string {
+  return bech32.encode(prefix, bech32.toWords(ripemd160(sha256(pubkey))));
+}
+
+/**
+ * Convert a secp256k1 compressed public key to an address
+ *
+ * @param {Uint8Array} pubkey The account's pubkey as base64 string, should be 33 bytes (compressed secp256k1)
+ * @param {String} [prefix="stride"] The address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function base64PubkeyToAddress(
+  pubkey: string,
+  prefix: string = "stride",
+): string {
+  return pubkeyToAddress(fromBase64(pubkey), prefix);
+}
+
+/**
+ * Convert self delegator address to validator address
+ *
+ * @param {String} selfDelegator The self delegator bech32 encoded address
+ * @param {String} [prefix="stride"] The self delegator address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function selfDelegatorAddressToValidatorAddress(
+  selfDelegator: string,
+  prefix: string = "stride",
+): string {
+  return bech32.encode(`${prefix}valoper`, bech32.decode(selfDelegator).words);
+}
+
+/**
+ * Convert self delegator address to validator address
+ *
+ * @param {String} validator The validator bech32 encoded address
+ * @param {String} [prefix="stride"] The self delegator address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function validatorAddressToSelfDelegatorAddress(
+  validator: string,
+  prefix: string = "stride",
+): string {
+  return bech32.encode(prefix, bech32.decode(validator).words);
+}
+
+/**
+ * Convert a Tendermint ed25519 public key to a consensus address
+ *
+ * @param {Uint8Array} pubkey The tendermint pubkey, should be 32 bytes (ed25519)
+ * @param {String} [prefix="stride"] The valcons address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the valcons account's address
+ */
+export function tendermintPubkeyToValconsAddress(
+  pubkey: Uint8Array,
+  prefix: string = "stride",
+): string {
+  return bech32.encode(
+    `${prefix}valcons`,
+    bech32.toWords(sha256(pubkey).slice(0, 20)),
+  );
+}
+
+/**
+ * Convert a secp256k1 compressed public key to an address
```

Comment:
```suggestion
 * Convert a base64 encoded public key to a validator address
```

Reply:
ccfafc9fd

Reply:
moved to stridejs https://github.com/Stride-Labs/stridejs/commit/2e3a23d